### PR TITLE
feat(server): Publicize the Timeouts struct for setting all timeouts at once.

### PR DESCRIPTION
**Pull request by reem**

### CODE REVIEW

**Comment by reem** [comment]
I am planning to use this in iron to expose keep alive and other timeout settings.

File: src/server/mod.rs [hunk 1]
```diff
@@ -147,11 +147,14 @@ pub struct Server<L = HttpListener> {
...
 
+/// A set of durations representing the various timeouts to be set on a server.
+///
+/// Note: more fields may be added in the future.
 #[derive(Clone, Copy, Debug)]
-struct Timeouts {
-    read: Option<Duration>,
-    write: Option<Duration>,
-    keep_alive: Option<Duration>,
+pub struct Timeouts {
```
Comment (untitaker) [inline src/server/mod.rs]: formatting nitpick

Comment (reem) [inline src/server/mod.rs]: what is the nitpick?

Comment (untitaker) [inline src/server/mod.rs]: Yikes, misread the diff. Sorry for the noise.

**Comment by seanmonstar** [comment]
I originally made that struct as a convenience when passing it around internally. I'm not so sure I'd want it part if a public API... Can you have a struct in Iron to do it?

**Comment by reem** [comment]
I could just duplicate this in iron, but it seemed better to just expose it.

**Comment by seanmonstar** [comment]
I'm not yet sure how timeouts will be set in the mio branch. Could you dupe
the struct in Iron for now? If this struct is kept in hyper, it can be made
public eventually, and you could swap the def for a reexport and it should
continue to work.

**Comment by reem** [comment]
Yeah, that's a good reason, I'll redefine it in iron.



Thanks,
         Jonathan Reem

**Comment by seanmonstar** [comment]
Closing this for now. Can revisit once hyper uses async io, and the timeout strategy is decided upon.

**Pull request closed without merging**